File: Laiterekisteri/sanitize_flows.py
import os
import json
import re

# Keys/fields commonly containing secrets in Power Automate / Logic Apps definitions
SECRET_KEYS = {
    "clientSecret", "secret", "refreshToken", "accessToken", 
    "clientId", "password", "authorization", "token", "sharedKey"
}

def sanitize_json(obj):
    if isinstance(obj, dict):
        new_dict = {}
        for k, v in obj.items():
            # If the key matches a secret name, sanitize the string value
            if any(secret_key.lower() in k.lower() for secret_key in SECRET_KEYS) and isinstance(v, str):
                new_dict[k] = "REDACTED"
            else:
                new_dict[k] = sanitize_json(v)
        return new_dict
    elif isinstance(obj, list):
        return [sanitize_json(item) for item in obj]
    elif isinstance(obj, str):
        # Regex to catch raw Azure AD / Google OAuth secrets if they appear in URL connection strings
        # E.g. client_secret=..., refresh_token=...
        cleaned = re.sub(r'(client_secret|refresh_token|code)=[^&"\']+', r'\1=REDACTED', obj)
        return cleaned
    return obj

count = 0
for root, _, files in os.walk("."):
    for file in files:
        if file.endswith(".json"):
            file_path = os.path.join(root, file)
            try:
                with open(file_path, 'r', encoding='utf-8') as f:
                    data = json.load(f)
                
                sanitized_data = sanitize_json(data)
                
                with open(file_path, 'w', encoding='utf-8') as f:
                    json.dump(sanitized_data, f, indent=2)
                
                count += 1
            except Exception as e:
                pass

print(f"Successfully sanitized JSON files in {count} locations!")

File: Laiterekisteri/sanitize_flows.ts
import * as fs from 'fs';
import * as path from 'path';

const SECRET_KEYS = new Set([
  'clientsecret', 'secret', 'refreshtoken', 'accesstoken',
  'clientid', 'password', 'authorization', 'token', 'sharedkey'
]);

function sanitizeJson(obj: any): any {
  if (Array.isArray(obj)) {
    return obj.map(sanitizeJson);
  } else if (obj !== null && typeof obj === 'object') {
    const newObj: Record<string, any> = {};
    for (const [key, val] of Object.entries(obj)) {
      if (SECRET_KEYS.has(key.toLowerCase()) && typeof val === 'string') {
        newObj[key] = 'REDACTED';
      } else {
        newObj[key] = sanitizeJson(val);
      }
    }
    return newObj;
  } else if (typeof obj === 'string') {
    return obj.replace(/(client_secret|refresh_token|code)=[^&"'\s]+/gi, '$1=REDACTED');
  }
  return obj;
}

function processDirectory(dir: string) {
  const entries = fs.readdirSync(dir, { withFileTypes: true });
  for (const entry of entries) {
    const fullPath = path.join(dir, entry.name);
    if (entry.name === 'node_modules' || entry.name === '.git') continue;

    if (entry.isDirectory()) {
      processDirectory(fullPath);
    } else if (entry.isFile() && entry.name.endsWith('.json')) {
      try {
        const content = fs.readFileSync(fullPath, 'utf8');
        const parsed = JSON.parse(content);
        const cleaned = sanitizeJson(parsed);
        fs.writeFileSync(fullPath, JSON.stringify(cleaned, null, 2), 'utf8');
        console.log(`Sanitized: ${fullPath}`);
      } catch (e) {
        // Skip non-valid JSON files
      }
    }
  }
}

processDirectory(process.cwd());
console.log('Sanitization complete!');